include/zephyr/kernel.h

```diff
@@ -545,9 +545,11 @@ __syscall int k_thread_join(struct k_thread *thread, k_timeout_t timeout);
  *
  * @param timeout Desired duration of sleep.
  *
- * @return Zero if the requested time has elapsed or if the thread was woken up
- * by the \ref k_wakeup call, the time left to sleep rounded up to the nearest
- * millisecond.
+ * @retval 0 Requested time has elapsed
+ * @retval >0 Number of milliseconds (rounded up) left to sleep if prematurely awakened
+ * @retval -1 Remaining time to sleep is too far into the future to represent
```

Comment:
Hm... Really needs to be INT_MAX, otherwise this is going to break preexisting code that was interpreting the value as defined.  Consider a loop like:

```c
while((rem = k_sleep(ms)) > 0) {
    ms -= rem;
}
```

Returning INT_MAX isn't numerically correct (because it can't be), but it recovers correct behavior from existing code.  We never defined the function to return an error case.

Also a nitpick that doesn't apply given the criticism above: this is clearly an error case and should probably be -EINVAL or

Reply:
1. I will change the name.
2. As to the cited example that the proposed change would break ... it is already broken. That is, if the value can not be represented in an int32_t, then the caller is going to get a garbage value back. That garbage value could be anywhere from INT32_MIN to INT32_MAX. To indicate an error, if you would prefer -EINVAL or something similar (I'm thinking -ERANGE) instead of -1, I'm okay with that too.